tests/testthat/test-seqCluster.R

```diff
@@ -0,0 +1,49 @@
+data(immdata)
+
+dist_test <- seqDist(immdata$data[1])
+
+
+# Arrange
+negative_test_cases <- list(
+  "Wrong data" = list(
+    args = list(
+      .data = immdata[1:2],
+      .dist = dist_test
+    )
+  ),
+  "No matching col" = list(
+    args = list(
+      .data = immdata$data$`A2-i131` %>% mutate("CDR3.nt" = NULL),
+      .dist = dist_test
+    )
+  ),
+  "Wrong samples" = list(
+    args = list(
+      .data = immdata$data[3],
+      .dist = dist_test
+    )
+  ),
+  "Multiple thresholds" = list(
+    args = list(
+      .data = immdata$data[1],
+      .dist = dist_test,
+      .fixed_threshold = 10,
+      .nt_similarity = 3
+    )
+  ),
+  "No thresholds" = list(
+    args = list(
+      .data = immdata$data[1],
+      .dist = dist_test,
+      .fixed_threshold = NULL
+    )
+  )
+)
+
+# Act
+
+negative_args <- map(negative_test_cases, "args")
+
+# Assert
+
+map2(names(negative_test_cases), negative_args, ~ test_that(.x, expect_error(do.call(seqCluster, .y))))
```

Comment:
Calling `seqCluster` is Act, and `test_that` is Assert. Please move the loop by test cases outside to make the entire Arrange-Act-Assert structure be inside the loop of test cases.

Reply:
`test_that` is used here to name the test, which is useful for analyzing error messages, so you can wrap `test_that` around AAA structure like this:
for (i in seq_along(negative_test_cases)) {
  test_that(names(negative_test_cases)[i],
    # Arrange
    ...
    # Act
    do.call(seqCluster, ...) %>%
    # Assert
      expect_error()
  )
}

Reply:
Errors cannot be handled via pipe, so the only way to use expect_error with negative tests is calling do.call inside expect_error(); otherwise, test will be stopped after receiving an error before expect_error check;

Also it is impossible to store test results in the variable, because test_that cannot handle it 
(you can try with something like 
```
tt<-expect_error(seqCluster('dddd'))
test_that('s',tt)
```
)
So, your change request are reasonable for positive tests, but negative tests cannot be done this way in test_that or I don't know how to do this.

Reply:
Then just write:
```
# Act, Assert
expect_error(do.call(...))

Reply:
```map2(names(negative_test_cases), negative_args, ~ test_that(.x, expect_error(do.call(seqCluster,.y)))) ```
is the same as you write above when test_that takes names of tests and check results is valid iterating by test cases, but without for cycle and using map2 instead which is preferred way in general, so this solution is valid;

Reply:
The point here is to decompose different parts of test and make it easier to understand and debug. For simple tests, map is ok; for tests that require extra arrangement, loop is preferred (or another structure that allows to write body of a parametrized test in clear way)